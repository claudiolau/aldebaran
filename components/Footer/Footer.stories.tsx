import React from "react";
import { Footer } from "./Footer";
import { ComponentMeta } from "@storybook/react";
import {
  Title,
  Subtitle,
  Description,
  Primary,
  ArgsTable,
  Stories,
  PRIMARY_STORY,
} from "@storybook/addon-docs";

export default {
  /* ðŸ‘‡ The title prop is optional.
   * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
  title: "Footer",
  component: Footer,
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Subtitle>A basic footer geared for personal websites.</Subtitle>
          <Description />
          <Primary />
          <ArgsTable story={PRIMARY_STORY} />
          <Stories />
        </>
      ),
    },
  },
} as ComponentMeta<typeof Footer>;

export const Basic = () => (
  <Footer topic={["Home", "Github", "Blog", "Gear", "Contact", "Github"]} />
);
