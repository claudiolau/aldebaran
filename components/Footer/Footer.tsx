import * as React from "react";
import styles from "./_footer.module.scss";

type FooterProps<T> = {
  topic: T[];
};

export const Footer: React.FC<FooterProps<String>> = ({ topic }) => {
  return (
    <>
      <div className={styles.footer}>
        {topic.map((item, index) => (
          <div key={index}>
            <a href={`${item}`}>{item}</a>
          </div>
        ))}
      </div>
    </>
  );
};
