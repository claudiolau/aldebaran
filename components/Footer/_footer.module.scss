.footer {
  position: relative;
  bottom: 2rem;
  display: grid;
  margin: auto;
  overflow: auto;
  padding: 4rem;
  max-width: 600px;
  grid-template-columns: repeat(3, 10rem);
  justify-content: center;
  row-gap: 2em;
}

@media only screen and (max-width: 600px) {
  .footer {
    grid-template-columns: repeat(2, 10rem);
    text-align-last: left;
  }
  .footerItem {
    place-items: center;
    text-align: center;
  }
}
