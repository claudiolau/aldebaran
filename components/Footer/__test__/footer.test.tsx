import * as React from "react";
import { render } from "@testing-library/react";
import { Footer } from "../Footer";

test("items navbar", async () => {
  const { asFragment } = render(<Footer topic={["test"]} />);
  expect(asFragment()).toMatchSnapshot();
});
