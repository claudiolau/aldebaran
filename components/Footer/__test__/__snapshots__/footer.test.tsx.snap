// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`items navbar 1`] = `
<DocumentFragment>
  <div>
    <div>
      <a
        href="test"
      >
        test
      </a>
    </div>
  </div>
</DocumentFragment>
`;
